docs/reference/text-structure/apis/find-structure-shared.asciidoc

```diff
@@ -0,0 +1,215 @@
+end::param-ecs-compatibility[]
+
+tag::param-has-header-row[]
+`has_header_row`::
+(Optional, Boolean) If you have set `format` to `delimited`, you can use this
+parameter to indicate whether the column names are in the first row of the text.
+If this parameter is not specified, the structure finder guesses based on the
+similarity of the first row of the text to other rows.
+end::param-has-header-row[]
+
+tag::param-line-merge-size-limit[]
+`line_merge_size_limit`::
+(Optional, unsigned integer) The maximum number of characters in a message when
+lines are merged to form messages while analyzing semi-structured text. The
+default is `10000`. If you have extremely long messages you may need to increase
+this, but be aware that this may lead to very long processing times if the way
+to group lines into messages is misdetected.
+end::param-line-merge-size-limit[]
+
+tag::param-lines-to-sample[]
+`lines_to_sample`::
+(Optional, unsigned integer) The number of lines to include in the structural
+analysis, starting from the beginning of the text. The minimum is 2; the default
+is `1000`. If the value of this parameter is greater than the number of lines in
+the text, the analysis proceeds (as long as there are at least two lines in the
+text) for all of the lines.
++
+--
+NOTE: The number of lines and the variation of the lines affects the speed of
+the analysis. For example, if you upload text where the first 1000 lines
+are all variations on the same message, the analysis will find more commonality
+than would be seen with a bigger sample. If possible, however, it is more
+efficient to upload sample text with more variety in the first 1000 lines than
+to request analysis of 100000 lines to achieve some variety.
+
+--
+end::param-lines-to-sample[]
+
+tag::param-quote[]
+`quote`::
+(Optional, string) If you have set `format` to `delimited`, you can specify the
+character used to quote the values in each row if they contain newlines or the
+delimiter character. Only a single character is supported. If this parameter is
+not specified, the default value is a double quote (`"`). If your delimited text
+format does not use quoting, a workaround is to set this argument to a character
+that does not appear anywhere in the sample.
+end::param-quote[]
+
+tag::param-should-trim-fields[]
+`should_trim_fields`::
+(Optional, Boolean) If you have set `format` to `delimited`, you can specify
+whether values between delimiters should have whitespace trimmed from them. If
+this parameter is not specified and the delimiter is pipe (`|`), the default
+value is `true`. Otherwise, the default value is `false`.
+end::param-should-trim-fields[]
+
+tag::param-timeout[]
+`timeout`::
+(Optional, <<time-units,time units>>) Sets the maximum amount of time that the
+structure analysis make take. If the analysis is still running when the timeout
+expires then it will be aborted. The default value is 25 seconds.
```

Comment:
```suggestion
(Optional, <<time-units,time units>>) Sets the maximum amount of time that the
structure analysis may take. If the analysis is still running when the timeout
expires then it will be stopped. The default value is 25 seconds.
```